### PR TITLE
fix db stats endpoint

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
## Summary of changes

Changes introduced in this pull request:
- added working ParityDb statistics endpoint,
- unified `enable_statistics` and `compression_type` params for databases,
- fixed stats endpoint header to accommodate current and all future db backends :brain: 

## Reference issue to close (if applicable)

Closes https://github.com/ChainSafe/forest/issues/2433

## Other information and links

Sample `ParityDb` stats output
<details>
  <summary>Log Output</summary>
  
  ```Paste log output here
❯ curl localhost:6116/stats/db
# DB statistics:
Column 0
Total values: 1373
Total bytes: 1551798
Total oversized values: 0
Total oversized bytes: 0
Total commits: 2303
New value insertions: 1373
Existing value insertions: 10048
Reference increases: 0
Missed reference increases: 0
Removals: 0
Missed removals: 0
Uncompressed bytes: 2896138
Compression deltas:
    2080-2111: 3432
    2112-2143: 2400
    2304-2335: 4986
    2336-2367: 3214
    2368-2399: 1932
    2432-2463: 3343
    2592-2623: 2201
    2752-2783: 7615
    2848-2879: 8241
    2912-2943: 2753
    3040-3071: 8802
    3104-3135: 2630
    3200-3231: 8746
    3232-3263: 2520
    3264-3295: 2440
    3296-3327: 2369
    3328-3359: 4833
    3520-3551: 2229
    3584-3615: 6787
    3616-3647: 2214
    3648-3679: 2113
    3680-3711: 4194
    3808-3839: 4181
    3840-3871: 6074
    4032-4063: 5908
    4096-4127: 32402
    4128-4159: 12398
    4192-4223: 8033
    4224-4255: 14701
    4288-4319: 60699
    4320-4351: 36805
    4352-4383: 6720
    4384-4415: 6516
    4416-4447: 6776
    4480-4511: 6928
    4512-4543: 2400
    4576-4607: 6932
    4640-4671: 4126
    4672-4703: 6954
    4736-4767: 7236
    4800-4831: 24133
    4896-4927: 7478
    4960-4991: 7418
    4992-5023: 18421
    5056-5087: 87884
    5088-5119: 91514
    5152-5183: 10318
    5248-5279: 34269
    5280-5311: 18815
    5312-5343: 37087
    5344-5375: 16438
    5376-5407: 3230
    5472-5503: 16932
    5504-5535: 11535
    5536-5567: 13916
    5568-5599: 3214
    5664-5695: 42721
    5696-5727: 2440
    5728-5759: 7696
    5760-5791: 36318
    5888-5919: 19634
    5920-5951: 9310
    5952-5983: 4843
    5984-6015: 2911
    6048-6079: 9009
    6080-6111: 4481
    6112-6143: 221647
    6496-6527: 5526
    6976-7007: 5908
    7168-7199: 6076
    7232-7263: 6112
    7296-7327: 6322
    7392-7423: 6340
    7456-7487: 6462
    7552-7583: 6476
    7616-7647: 6602
    7648-7679: 6516
    7712-7743: 6720
    7776-7807: 6776
    7936-7967: 6928
    8064-8095: 6932
    8160-8191: 6954
    8352-8383: 7236
    8448-8479: 21932
    8640-8671: 7478
    8672-8703: 7418
    8896-8927: 7684
    9120-9151: 7918
    9312-9343: 8088
    9408-9439: 16398
    9504-9535: 8278
    9664-9695: 16756
    9728-9759: 8524
    10336-10367: 9036
    10592-10623: 9310
Queries per size tier: [5053, 0, 0, 0, 10953, 5585, 1, 0, 28, 0, 6, 9, 43, 1, 1, 0, 0, 1, 11, 23, 0, 0, 0, 8, 0, 0, 1, 1, 6, 0, 29, 0, 12, 1030, 250, 15, 3, 8, 27, 0, 0, 0, 161, 0, 182, 42, 8, 6197, 1528, 147, 43, 21, 23, 0, 1, 81, 4, 44, 160, 58, 336, 19, 225, 8, 261, 35, 4, 33, 277, 120, 64, 11, 41, 117, 23, 193, 17, 872, 9, 22, 30, 103, 5972, 30, 63, 1, 9, 1, 379, 142, 513, 64, 19, 0, 0, 1, 28, 82, 264, 2, 19, 0, 5, 0, 844, 106, 0, 0, 19, 1, 0, 160, 61, 0, 0, 0, 14, 93, 28, 1, 2, 31, 57, 0, 910, 3725, 28, 6028, 14, 6877, 5246, 3210, 104, 1573, 566, 107, 19, 0, 3, 2444, 0, 0, 30, 0, 0, 0, 4, 6, 1, 7, 1, 0, 32, 318, 195, 4, 7, 5, 361, 609, 67, 111, 40, 38, 196, 189, 93, 47, 83, 36, 66, 194, 3, 18, 34, 57, 49, 53, 70, 2, 38, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 24]
Missed queries: 132
Value histogram:
    0-31: 125
    32-63: 4
    64-95: 30
    96-127: 1
    128-159: 30
    160-191: 1
    192-223: 12
    224-255: 36
    256-287: 29
    288-319: 4
    320-351: 205
    384-415: 3
    416-447: 58
    480-511: 2
    512-543: 68
    608-639: 31
    672-703: 1
    704-735: 5
    736-767: 34
    800-831: 2
    832-863: 56
    896-927: 5
    1024-1055: 7
    1056-1087: 8
    1088-1119: 6
    1216-1247: 27
    1376-1407: 54
    1504-1535: 27
    1760-1791: 1
    1792-1823: 2
    1920-1951: 29
    1952-1983: 1
    1984-2015: 2
    2048-2079: 2
    2080-2111: 3
    2112-2143: 1
    2176-2207: 3
    2208-2239: 1
    2272-2303: 1
    2304-2335: 29
    2336-2367: 4294967295
    2368-2399: 7
    2400-2431: 2
    2432-2463: 4294967295
    2464-2495: 1
    2496-2527: 8
    2528-2559: 17
    2560-2591: 3
    2592-2623: 28
    2656-2687: 1
    2688-2719: 1
    2720-2751: 4
    2752-2783: 4294967293
    2784-2815: 1
    2816-2847: 2
    2848-2879: 4294967295
    2912-2943: 2
    3040-3071: 4294967293
    3104-3135: 4294967295
    3200-3231: 4294967294
    3232-3263: 4294967295
    3264-3295: 4294967295
    3296-3327: 4294967295
    3328-3359: 4294967294
    3360-3391: 19
    3520-3551: 4
    3584-3615: 4294967293
    3616-3647: 4294967295
    3648-3679: 4294967295
    3680-3711: 4294967294
    3744-3775: 14
    3808-3839: 4294967294
    3840-3871: 4294967293
    4032-4063: 4294967294
    4096-4127: 18
    4128-4159: 4294967292
    4192-4223: 4294967293
    4224-4255: 4294967291
    4288-4319: 26
    4320-4351: 14
    4352-4383: 4294967294
    4384-4415: 4294967294
    4416-4447: 4294967294
    4480-4511: 4294967294
    4512-4543: 1
    4576-4607: 4294967294
    4640-4671: 2
    4672-4703: 4294967294
    4736-4767: 4294967294
    4800-4831: 4294967291
    4896-4927: 4294967294
    4960-4991: 4294967294
    4992-5023: 8
    5056-5087: 33
    5088-5119: 39
    5152-5183: 4294967295
    5248-5279: 14
    5280-5311: 4294967293
    5312-5343: 15
    5344-5375: 4294967292
    5376-5407: 1
    5472-5503: 4294967292
    5504-5535: 4294967295
    5536-5567: 5
    5568-5599: 1
    5664-5695: 20
    5696-5727: 1
    5728-5759: 3
    5760-5791: 15
    5888-5919: 7
    5920-5951: 4294967294
    5952-5983: 2
    5984-6015: 1
    6048-6079: 4
    6080-6111: 2
    6112-6143: 98
    6496-6527: 2
    6976-7007: 2
    7168-7199: 2
    7232-7263: 2
    7296-7327: 2
    7392-7423: 2
    7456-7487: 2
    7552-7583: 2
    7616-7647: 2
    7648-7679: 2
    7712-7743: 2
    7776-7807: 2
    7936-7967: 2
    8064-8095: 2
    8160-8191: 2
    8352-8383: 2
    8448-8479: 6
    8640-8671: 2
    8672-8703: 2
    8896-8927: 2
    9120-9151: 2
    9312-9343: 2
    9408-9439: 4
    9504-9535: 2
    9664-9695: 4
    9728-9759: 2
    10336-10367: 2
    10592-10623: 2
  ```
</details> 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: node/db/src/parity_db.rs [hunk 1]
```diff
@@ -17,6 +18,7 @@ use std::sync::Arc;
 #[derive(Clone)]
 pub struct ParityDb {
     pub db: Arc<parity_db::Db>,
+    statistcs_enabled: bool,
```
Comment (elmattic) [inline node/db/src/parity_db.rs:21]: Some small typo here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,9 +2,12 @@
 Notable updates:
 
 ### Added
+* [database] added ParityDb statistics to the stats endpoint [#2444](https://github.com/ChainSafe/forest/pull/2444).
+
 
 ### Changed
 * [libp2p] Ban peers with duration. Banned peers are automatically unbanned after a period of 1h.: [#2396](https://github.com/ChainSafe/forest/pull/2396)
+* [database] ParityDb fields `stats` and `compression` were renamed to `enable_statistics` and `compression_type` respectively. [#2444](https://github.com/ChainSafe/forest/pull/2444).
```
Comment (elmattic) [inline CHANGELOG.md:10]: What do you think of:

"* [config] `stats` and `compression` keys in `parity_db` section were renamed to `enable_statistics` and `compression_type` respectively."

**Review by elmattic — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (LesnyRumcajs) [reply CHANGELOG.md]: Sounds good

File: node/db/src/parity_db.rs (hunk 1 above)
Comment (LesnyRumcajs) [reply node/db/src/parity_db.rs]: Good catch!

**Review by lemmih — APPROVED** [review]

**Pull request merged**